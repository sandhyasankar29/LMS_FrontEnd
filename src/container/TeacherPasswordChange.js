import { React, useState } from 'react';
import axios from 'axios';
import {
    Box,
    Button,
    Grommet,
    TextInput,
    Image,
    Anchor,
    Header,
    Nav,
    Text,
    Heading,
    Layer,
    Form,
    FormField,
    MaskedInput,

} from 'grommet';
import { Hide, View, User } from 'grommet-icons';
import passImage from '../assets/teacherpass.svg'
import LogOut from './LogOut';
import { grommet } from 'grommet/themes';
import Swal from 'sweetalert2'
import './GlobalVariables'
const TeacherPasswordChange = () => {
    const urlChange = 'http://'+{global,ip}.ip+':3001/changepassword';
    const urlVerify = 'http://'+{global,ip}.ip+':3001/verifyteacherotp';
    const Toast = Swal.mixin({
        toast: true,
        position: 'top-end',
        showConfirmButton: false,
        timer: 3000,
        timerProgressBar: true,
        didOpen: (toast) => {
            toast.addEventListener('mouseenter', Swal.stopTimer)
            toast.addEventListener('mouseleave', Swal.resumeTimer)
        }
    })
    const [password, setPassword] = useState("")
    const [newpass, setPass] = useState('')
    const [email, setEmail] = useState("")
    const [reveal, setReveal] = useState(false);
    const [open, setOpen] = useState(false);
    const [otp, setOtp] = useState();
    const items = [
        { label: 'Dashboard', href: '/teacherdashboard', value: 0 },
        { label: 'Password Change', href: '/teacherchangepassword', value: 1 },
        { label: 'Button', href: '#', value: 2 },
    ];
    const onSubmitPass = () => {
        if (email && password) {
            axios.post(urlChange, {
                tuserid: localStorage.getItem('tuserid'),
                password: password,
                email: email,
            }).then(res => {
                setOpen(true)
                setEmail('')
                setPassword('')
            }).catch(error => {
                Swal.fire({
                    icon: 'error',
                    title: 'Oops...',
                    text: 'Please Verify the Credentials',
                })
            })
        }
    }
    const onOTPSubmit = () => {
        setOpen(false)
        if (newpass && otp) {
            axios.post(urlVerify, {
                tuserid: localStorage.getItem('tuserid'),
                password: newpass,
                otp: otp
            }).then(res => {
                Toast.fire({
                    icon: 'success',
                    title: 'Password changed'
                })
            }).catch(error => {
                Toast.fire({
                    icon: 'error',
                    title: 'OTP Wrong'
                })
            })
        }
    }
    return (
        <Grommet theme={grommet}>
            <Header background="dark-1" pad="small">
                <Box direction="row" align="center" gap="small">
                    <User />
                    <Anchor color="white" href="#">
                        Welcome {localStorage.getItem('tuserid')}
                    </Anchor>
                </Box>
                <Nav direction="row">
                    {items.map(item => (
                        item.value !== 2 ? <Anchor href={item.href} label={item.label} key={item.label} /> : <LogOut route={'/teachersignin'} />
                    ))}
                </Nav>
            </Header>
            <Box style={{ marginTop: "7%" }} fill align="center" justify="center">
                <Box height="small" width="large" style={{
                    marginBottom: "3%",
                }}>
                    <Image src={passImage} fit="contain" />
                </Box>
                <Form>
                    <Box width="medium" justify="center" >
                        <FormField label="Email" name="email">
                            <MaskedInput
                                name="email"
                                placeholder='example@gmail.com'
                                value={email}
                                onChange={event => setEmail(event.target.value)}
                            />
                        </FormField>
                        <FormField label="Old Password" name="passsword">
                            <Box direction="row" justify="between" margin={{ top: 'xxsmall' }}>
                                <TextInput
                                    plain
                                    type={reveal ? 'text' : 'password'}
                                    value={password}
                                    onChange={event => setPassword(event.target.value)}
                                />
                                <Button
                                    icon={reveal ? <View size="medium" /> : <Hide size="medium" />}
                                    onClick={() => setReveal(!reveal)}
                                />
                            </Box>
                        </FormField>
                        <Box direction="row" justify="center" margin={{ top: 'medium' }}>
                            <Button data-testid="button" active={true} type="submit" label="Request Password Change" onClick={onSubmitPass} primary />
                        </Box>
                    </Box>
                </Form>
                {open ? (
                    <Layer position="center">
                        <Box pad="medium" gap="small" width="medium">
                            <Heading level={3} margin="none">
                                MOBILE OTP VERIFICATION
                            </Heading>
                            <Text>Enter the OTP</Text>
                            <TextInput
                                name="otp"
                                value={otp}
                                onChange={event => setOtp(event.target.value)}
                                placeholder='One Time Pass'
                            />
                            <TextInput
                                name="newpass"
                                value={newpass}
                                onChange={event => setPass(event.target.value)}
                                placeholder='Enter Preferred Password'
                            />
                            <Box
                                as="footer"
                                gap="small"
                                direction="row"
                                align="center"
                                justify="end"
                                pad={{ top: 'medium', bottom: 'small' }}
                            >
                                <Button label="Submit" onClick={onOTPSubmit} color="dark-3" />
                                <Button
                                    label={
                                        <Text color="white">
                                            <strong>Cancel</strong>
                                        </Text>
                                    }
                                    onClick={() => { setOpen(false) }}
                                    primary
                                    color="status-critical"
                                />
                            </Box>
                        </Box>
                    </Layer>
                ) : null}
            </Box>
        </Grommet>
    );
};

export default TeacherPasswordChange;