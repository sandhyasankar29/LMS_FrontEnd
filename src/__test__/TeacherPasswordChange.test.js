import React from 'react';
import ReactDOM from 'react-dom';
import TeacherPasswordChange from '../container/TeacherPasswordChange';
import { cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'

afterEach(cleanup)
it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<TeacherPasswordChange></TeacherPasswordChange>, div)
})

it("matches snapshot", () => {
    const tree = renderer.create(<TeacherPasswordChange></TeacherPasswordChange>).toJSON();
    expect(tree).toMatchSnapshot();
})
