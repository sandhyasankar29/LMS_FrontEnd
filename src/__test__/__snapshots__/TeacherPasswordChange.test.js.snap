// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 jirFMY"
>
  <header
    className="StyledBox-sc-13pk1d4-0 jwhdBG"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 dfYOnT"
    >
      <svg
        aria-label="User"
        className="StyledIcon-ofa7kd-0 dcacLZ"
        viewBox="0 0 24 24"
      >
        <path
          d="M8,24 L8,19 M16,24 L16,19 M3,24 L3,19 C3,14.0294373 7.02943725,11 12,11 C16.9705627,11 21,14.0294373 21,19 L21,24 M12,11 C14.7614237,11 17,8.76142375 17,6 C17,3.23857625 14.7614237,1 12,1 C9.23857625,1 7,3.23857625 7,6 C7,8.76142375 9.23857625,11 12,11 Z"
          fill="none"
          stroke="#000"
          strokeWidth="2"
        />
      </svg>
      <div
        className="StyledBox__StyledBoxGap-sc-13pk1d4-1 eBVJAG"
      />
      <a
        className="StyledAnchor-sc-1rp7lwl-0 ePWGdF"
        href="#"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        Welcome 
      </a>
    </div>
    <div
      className="StyledBox__StyledBoxGap-sc-13pk1d4-1 ffHQai"
    />
    <nav
      className="StyledBox-sc-13pk1d4-0 glIhNO"
    >
      <a
        className="StyledAnchor-sc-1rp7lwl-0 eaoGTv"
        href="/teacherdashboard"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        Dashboard
      </a>
      <div
        className="StyledBox__StyledBoxGap-sc-13pk1d4-1 ffHQai"
      />
      <a
        className="StyledAnchor-sc-1rp7lwl-0 eaoGTv"
        href="/teacherchangepassword"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        Password Change
      </a>
      <div
        className="StyledBox__StyledBoxGap-sc-13pk1d4-1 ffHQai"
      />
      <button
        className="StyledButton-sc-323bzc-0 bZMlsb"
        id="logoutBtn"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        type="button"
      >
        <svg
          aria-label="Logout"
          className="StyledIcon-ofa7kd-0 eBBIEF"
          viewBox="0 0 24 24"
        >
          <path
            d="M13,9 L13,2 L1,2 L1,22 L13,22 L13,15 M22,12 L5,12 M17,7 L22,12 L17,17"
            fill="none"
            stroke="#231F20"
            strokeWidth="2"
          />
        </svg>
      </button>
    </nav>
  </header>
  <div
    className="StyledBox-sc-13pk1d4-0 bxevCE"
    style={
      Object {
        "marginTop": "7%",
      }
    }
  >
    <div
      className="StyledBox-sc-13pk1d4-0 bJtzEt"
      style={
        Object {
          "marginBottom": "3%",
        }
      }
    >
      <img
        className="StyledImage-ey4zx9-0 iBeQsm"
        src="teacherpass.svg"
      />
    </div>
    <form
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="StyledBox-sc-13pk1d4-0 bUdAeb"
      >
        <div
          className="StyledBox-sc-13pk1d4-0 dLXLoO FormField__FormFieldBox-m9hood-0 jGEMXj"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <label
            className="StyledText-sc-1sadyjn-0 drTEXR"
          >
            Email
          </label>
          <div
            className="StyledBox-sc-13pk1d4-0 gmReJc FormField__FormFieldContentBox-m9hood-1"
          >
            <div
              className="StyledMaskedInput__StyledMaskedInputContainer-sc-99vkfa-1 dOerlP"
            >
              <input
                autoComplete="off"
                className="StyledMaskedInput-sc-99vkfa-0 gMZsTd"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="example@gmail.com"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="StyledBox-sc-13pk1d4-0 dLXLoO FormField__FormFieldBox-m9hood-0 jGEMXj"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <label
            className="StyledText-sc-1sadyjn-0 drTEXR"
          >
            Old Password
          </label>
          <div
            className="StyledBox-sc-13pk1d4-0 gmReJc FormField__FormFieldContentBox-m9hood-1"
          >
            <div
              className="StyledBox-sc-13pk1d4-0 hpZuiW"
            >
              <div
                className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 dpaJQp"
              >
                <input
                  autoComplete="off"
                  className="StyledTextInput-sc-1x30a0s-0 hubMyb"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  type="password"
                  value=""
                />
              </div>
              <button
                className="StyledButton-sc-323bzc-0 hiyJPE"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                type="button"
              >
                <svg
                  aria-label="Hide"
                  className="StyledIcon-ofa7kd-0 dEeaTZ"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12,17 C9.27272727,17 6,14.2222222 6,12 C6,9.77777778 9.27272727,7 12,7 C14.7272727,7 18,9.77777778 18,12 C18,14.2222222 14.7272727,17 12,17 Z M11,12 C11,12.55225 11.44775,13 12,13 C12.55225,13 13,12.55225 13,12 C13,11.44775 12.55225,11 12,11 C11.44775,11 11,11.44775 11,12 Z M20,5 L4,19"
                    fill="none"
                    stroke="#000"
                    strokeWidth="2"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          className="StyledBox-sc-13pk1d4-0 cgsQvy"
        >
          <button
            className="StyledButton-sc-323bzc-0 iVPikU"
            data-testid="button"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            type="submit"
          >
            Request Password Change
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
